cdap-docs/developers-manual/source/getting-started/standalone/docker.rst

```diff
@@ -14,9 +14,128 @@ Docker is one of the easiest ways to start working with CDAP without having to m
+To use the **Docker image**, you can either start the container from :ref:`a command line
+<docker-command-line>` or use Docker's :ref:`Kitematic <docker-kitematic>` (on Mac OS X
+and Windows), a graphical user interface for running Docker containers.
+
+.. _docker-command-line:
+
+Docker from a Command Line
+==========================
+
+Docker environments are available for a variety of platforms. Download and install Docker for your platform by
+following the `platform-specific installation instructions <https://docs.docker.com/engine/installation/>`__
+from `Docker.com <https://docker.com>`__, and verify that the Docker environment is working and has
+started correctly.
+  
+#. If you are not running on Linux, you will need to create and start a Docker Virtual Machine (VM) before you
+   can use containers. For example:
+   
+   .. tabbed-parsed-literal::
+     :tabs: "Mac OS X",Windows
+     :mapping: linux,windows
+     :dependent: linux-windows
+     :languages: console,shell-session
+     
+     $ docker-machine create --driver virtualbox cdap
+     $ docker-machine env cdap
+     
+   This will create a new Docker virtual machine using VirtualBox named ``cdap`` and print out the environment.
+
+#. When you run ``docker-machine env cdap``, it will print a message on the screen such as::
+
+     export DOCKER_TLS_VERIFY="1"
+     export DOCKER_HOST="tcp://192.168.99.100:2376"
+     export DOCKER_CERT_PATH="/Users/<username>/.docker/machine/machines/cdap"
+     export DOCKER_MACHINE_NAME="cdap"
+     # Run this command to configure your shell: 
+     # eval $(docker-machine env cdap)
+ 
+   It is essential to run these export commands (or the single ``eval`` command). Otherwise,
+   subsequent Docker commands will fail because they won't be able to connect to the correct
+   Docker VM.
+  
+#. If you are running **Docker on either Mac OS X or Microsoft Windows**, Docker is running a
+   virtual Linux machine on top of your host OS. You will need to use the address shown
+   above (such as ``192.168.99.100``) as the host name when either connecting to the CDAP
+   UI or making an HTTP request.
+
+#. Once Docker has started, pull down the *CDAP Docker Image* from the Docker Hub using:
+
+   .. tabbed-parsed-literal::
+     :tabs: "Linux or Mac OS X",Windows
+     :mapping: linux,windows
+     :dependent: linux-windows
+     :languages: console,shell-session
+ 
+     .. Linux or Mac OS X
+     
+     $ docker pull caskdata/cdap-standalone:|release|
+
+     .. Windows
+    
+     > docker pull caskdata/cdap-standalone:|release|
+
+#. Start the *CDAP Standalone Docker container* with:
+
+   .. tabbed-parsed-literal::
+     :tabs: "Linux or Mac OS X",Windows
+     :mapping: linux,windows
+     :dependent: linux-windows
+     :languages: console,shell-session
+ 
+     $ docker run --detach --name cdap-standalone -p 9999:9999 -p 10000:10000 caskdata/cdap-standalone:|release|
```

Comment:
I'd like to keep the docs as close as possible to the docs seen on Docker Hub, which invariably use `-d` versus the long form. Also, less typing means less typos.